app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -82,148 +133,168 @@ const BigDelegatorOffer = () => (
+  )
+  const [isTicker, setIsTicker] = useState(false)
+  const [isPoolHash, setIsPoolHash] = useState(!!ADALITE_CONFIG.ADALITE_STAKE_POOL_ID)
 
-  componentDidMount() {
-    if (ADALITE_CONFIG.ADALITE_STAKE_POOL_ID !== '') this.props.selectAdaliteStakepool()
+  const handleOnInput = (event: any): void => {
+    const fieldValue: string = event && event.target && event.target.value
+    const isTicker = fieldValue && !!validStakepools.getPoolInfoByTicker(fieldValue)
+    const isPoolHash = fieldValue && !!validStakepools.getPoolInfoByPoolHash(fieldValue)
+    setFieldValue(fieldValue)
+    setIsTicker(isTicker)
+    setIsPoolHash(isPoolHash)
+    const poolHash = isTicker
+      ? validStakepools.getPoolInfoByTicker(fieldValue).poolHash
+      : fieldValue
+    const hasTickerMapping = validStakepools.hasTickerMapping
+    const error =
+      poolHash === '' || validStakepools.getPoolInfoByPoolHash(poolHash)
+        ? null
+        : {code: 'InvalidStakepoolIdentifier', params: {hasTickerMapping}}
+    updateStakePoolIdentifier(poolHash, error)
   }
 
-  render({
-    updateStakePoolIdentifier,
-    stakePool,
-    delegationFee,
-    calculatingDelegationFee,
-    delegationValidationError,
-    isShelleyCompatible,
-    confirmTransaction,
-    shouldShowConfirmTransactionDialog,
-    txSuccessTab,
-    gettingPoolInfo,
-    poolRecommendation,
-    pool,
-    isBigDelegator,
-    withAccordion,
-    title,
-  }) {
-    const delegationHandler = async () => {
-      await confirmTransaction('delegate')
-    }
-    const validationError =
-      delegationValidationError || stakePool.validationError || stakePool.poolHash === ''
-
-    const delegationHeader = <h2 className="card-title no-margin">{title}</h2>
-    const delegationContent = (
-      <Fragment>
-        <div className="stake-pool">
-          <ul className="stake-pool-list">
-            <li className="stake-pool-item">
-              {isBigDelegator && <BigDelegatorOffer />}
-              <div
-                className={`input stake-pool-id delegation-input-row ${
-                  this.state.dropShadow ? 'focus' : ''
-                }`}
-              >
-                <div className="delegation-input-label">Stake Pool ID:</div>
-                <input
-                  type="text"
-                  className="delegation-input"
-                  name={'pool'}
-                  value={stakePool.poolHash}
-                  onInput={updateStakePoolIdentifier}
-                  autoComplete="off"
-                  onFocus={() => this.setState({dropShadow: true})}
-                  onBlur={() => this.setState({dropShadow: false})}
-                />
-              </div>
-              <StakePoolInfo pool={stakePool} gettingPoolInfo={gettingPoolInfo} />
-              <div />
-            </li>
-          </ul>
-        </div>
+  const delegationHandler = async (): Promise<void> => {
+    await confirmTransaction('delegate')
```

Comment:
is this rebased? I think TxType.DELEGATE can already be used

Reply:
I chcked `confirmTransaction` function and it's logic is checking against strings, not `TxType` type. I don't want to refactor that in this PR, it's already big enough.